homeassistant/components/decora_wifi/config_flow.py

```diff
@@ -0,0 +1,119 @@
+"""Config flow for myLeviton decora_wifi."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .common import CommFailed, DecoraWifiPlatform, LoginFailed
+from .const import CONF_TITLE, DOMAIN
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class DecoraWifiConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Decora Wifi config flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self.session = None
+        self.data = {}
+        self.entry = None
+
+    # Disable pylint error about number of arguments in step_user
+    # pylint: disable=arguments-differ
+    async def async_step_user(self, user_input=None, errors=None):
+        """Prompt for user input to setup decora_wifi."""
+
+        data_schema = {
+            vol.Required(CONF_USERNAME): str,
+            vol.Required(CONF_PASSWORD): str,
+        }
+
+        if user_input is None:
+            # Show the form requesting the Username and Password for Decora Wifi
+            return self.async_show_form(
+                step_id="user", data_schema=vol.Schema(data_schema)
+            )
+
+        # Update flow state
+        self.data.update(user_input)
+
+        # Shouldn't need errors anymore after this point.
+        errors = {}
+
+        # Get user from existing entry and abort if already setup.
+        self.entry = await self.async_set_unique_id(self.data[CONF_USERNAME])
+        self._abort_if_unique_id_configured()
+
+        try:
+            # Attempt to log in with the credentials provided by the user.
+            self.session = await DecoraWifiPlatform.async_setup_decora_wifi(
+                self.hass,
+                email=self.data[CONF_USERNAME],
+                password=self.data[CONF_PASSWORD],
+            )
+        except LoginFailed:
+            errors["base"] = "invalid_auth"
```

Comment:
Please make this more DRY

Let it fall through and show the form again if errors exist do you don't have to repeat yourself with `async_show_form`